delete_purchases/delete_purchases.py

```diff
@@ -39,51 +28,83 @@ def get_purchase_lines(api):
+            if any(part_inventory["installed"] for part_inventory in purchase_order_line["node"]["partInventories"]) or any(part_inventory["kitted"] for part_inventory in purchase_order_line["node"]["partInventories"])or any(part_inventory["received"] for part_inventory in purchase_order_line["node"]["partInventories"]):
+                po_id = purchase_order_line["node"]["purchaseOrder"]["id"]
+                list_of_pos.add(po_id)
+    return list(list_of_pos)
+
+def build_list_receipts(receipts):
+    LIST_OF_POS = []
+    for receipt in receipts["receipts"]["edges"]:
+        polines = receipt["node"]["purchaseOrderLines"]
+        for poline in polines:
+            poline_id = poline["id"]
+            LIST_OF_POS.append(poline_id)
+    return LIST_OF_POS
 
 def delete_receipts(receipts, api):
     for receipt in receipts["receipts"]["edges"]:
         receipt_id = receipt["node"]["id"]
-        logger.info(f'Deleting receipt id: {receipt_id}')
+        print('Deleting receipt id: ',receipt_id)
         etag = receipt["node"]["_etag"]
         request_body = {
             "query": queries.DELETE_RECEIPT,
             "variables": {"id": receipt_id, "etag": etag},
         }
         api.request(request_body)
 
-
 def delete_purchase_lines(purchase_lines, api):
+    LIST_POS = set()
+    POS_WITH_ABOMITEMS_TO_SKIP = build_list_aboms_items(purchase_lines)
     for purchase_line in purchase_lines["purchaseOrderLines"]["edges"]:
+        po_id = purchase_line["node"]["purchaseOrder"]["id"]
+        po_status =  purchase_line["node"]["purchaseOrder"]["status"]
         purchase_line_id = purchase_line["node"]["id"]
-        logger.info(f'Deleting purchase line id: {purchase_line_id}')
-        etag = purchase_line["node"]["_etag"]
+        etag=get_purchase_line_etag(purchase_line_id,api)
+        if po_id in POS_WITH_ABOMITEMS_TO_SKIP or po_status == 'CANCELED' or po_status == 'RECEIVED':
+            print("Skipping PO LINE",purchase_line_id)
+            LIST_POS.add(po_id)
+            continue
+        print("Deleting purchase line id",purchase_line_id)
         request_body = {
             "query": queries.DELETE_PURCHASE_LINE,
             "variables": {"id": purchase_line_id, "etag": etag},
         }
         api.request(request_body)
+    return list(LIST_POS)
 
-
-def delete_purchases(purchases, api):
+def delete_purchases(pos_to_save,purchases, api):
+    POS_WITH_RECEIPTS_TO_SKIP = build_list_receipts(receipts)
     for purchase in purchases["purchaseOrders"]["edges"]:
         purchase_id = purchase["node"]["id"]
-        if purchase_id in PURCHASES_TO_SKIP or purchase['node']['approvals']:
-            logger.info(f'Skipping purchase id: {purchase_id}')
-            continue
-        logger.info(f'Deleting purchase id: {purchase_id}')
         etag = purchase["node"]["_etag"]
+        if purchase_id in PURCHASES_TO_SKIP or purchase_id in POS_WITH_RECEIPTS_TO_SKIP or purchase_id in pos_to_save:
```

Comment:
Why do we need a separate list for `POS_WITH_RECEIPTS_TO_SKIP`? Can it just not be added to `PURCHASES_TO_SKIP`?